dotnet/src/SFTPService/SftpService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.Extensions.Logging;
+using Renci.SshNet;
+using Renci.SshNet.Sftp;
+
+namespace SFTPService
+{
+    /// <summary>
+    /// Class containing sftp file operations
+    /// </summary>
+    public class SftpService : ISftpService
+    {
+        private readonly ILogger<SftpService> _logger;
+        private readonly SftpConfig _config;
+
+        public SftpService(ILogger<SftpService> logger, SftpConfig sftpConfig)
+        {
+            _logger = logger;
+            _config = sftpConfig;
+        }
+
+        /// <summary>
+        /// Get list of all the files in remote directory
+        /// </summary>
+        /// <param name="remoteDirectory">is the directory path.</param>
+        /// <returns>IEnumerable<SftpFile> , containing collection of files in the given directory</returns>
+       
+        public IEnumerable<SftpFile> ListAllFiles(string remoteDirectory = ".")
+        {
+            using var client = new SftpClient(_config.Host, _config.Port == 0 ? 22 : _config.Port, _config.UserName, _config.Password);
+            try
+            {
+                client.Connect();
+                return client.ListDirectory(remoteDirectory);
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, $"Failed in listing files under [{remoteDirectory}]");
+                return null;
+            }
+            finally
+            {
+                client.Disconnect();
+            }
+        }
+
+        /// <summary>
+        /// Upload the files from a local directory to the remote directory
+        /// </summary>
+        /// <param name="localFilePath">is the source path of the file .</param>
+        /// <param name="remoteFilePath">is the destination path of the file .</param>
+        /// <returns>void</returns>
+        public void UploadFile(string localFilePath, string remoteFilePath)
+        {           
+            using var client = new SftpClient(_config.Host, _config.Port == 0 ? 22 : _config.Port, _config.UserName, _config.Password);
+            try
+            {
+                client.Connect();
+                using var s = File.OpenRead(localFilePath);
+                client.UploadFile(s, remoteFilePath);
+                _logger.LogInformation($"Finished uploading file [{localFilePath}] to [{remoteFilePath}]");
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, $"Failed in uploading file [{localFilePath}] to [{remoteFilePath}]");
```

Comment:
@paras-lalchandani `Failed to upload file`

Reply:
@motymichaely done